test/EFCore.Specification.Tests/Query/NorthwindIncludeQueryTestBase.cs

```diff
@@ -3785,6 +3904,139 @@ public virtual async Task NoTracking_Include_with_cycles_does_not_throw_when_per
+        {
+            var options = new Newtonsoft.Json.JsonSerializerSettings
+            {
+                ReferenceLoopHandling = Newtonsoft.Json.ReferenceLoopHandling.Ignore,
+                EqualityComparer = LegacyReferenceEqualityComparer.Instance,
+                Formatting = writeIndented
+                    ? Newtonsoft.Json.Formatting.Indented
+                    : Newtonsoft.Json.Formatting.None
+            };
+
+           var serializeObject = Newtonsoft.Json.JsonConvert.SerializeObject(collection, options);
+
+           return Newtonsoft.Json.JsonConvert.DeserializeObject<T>(serializeObject);
+        }
+
+        private static void VerifyCustomer(NorthwindContext context, Customer customer, Dictionary<string, Customer> customersMap)
+        {
+            var trackedCustomer = context.Customers.Find(customer.CustomerID);
+            Assert.Equal(trackedCustomer.Address, customer.Address);
+            Assert.Equal(trackedCustomer.City, customer.City);
+            Assert.Equal(trackedCustomer.Country, customer.Country);
+            Assert.Equal(trackedCustomer.Fax, customer.Fax);
+            Assert.Equal(trackedCustomer.Phone, customer.Phone);
+            Assert.Equal(trackedCustomer.Region, customer.Region);
+            Assert.Equal(trackedCustomer.CompanyName, customer.CompanyName);
+            Assert.Equal(trackedCustomer.ContactName, customer.ContactName);
+            Assert.Equal(trackedCustomer.ContactTitle, customer.ContactTitle);
+            Assert.Equal(trackedCustomer.IsLondon, customer.IsLondon);
+            Assert.Equal(trackedCustomer.PostalCode, customer.PostalCode);
+            Assert.Equal(trackedCustomer.CustomerID, customer.CustomerID);
+
+            if (customersMap != null)
+            {
+                if (customersMap.TryGetValue(customer.CustomerID, out var mappedCustomer))
+                {
+                    Assert.Same(customer, mappedCustomer);
+                }
+
+                customersMap[customer.CustomerID] = customer;
+            }
+        }
+
+        private static void VerifyOrder(NorthwindContext context, Order order, IDictionary<int, Order> ordersMap)
+        {
+            var trackedOrder = context.Orders.Find(order.OrderID);
+            Assert.Equal(trackedOrder.Freight, order.Freight);
+            Assert.Equal(trackedOrder.OrderDate, order.OrderDate);
+            Assert.Equal(trackedOrder.RequiredDate, order.RequiredDate);
+            Assert.Equal(trackedOrder.ShipAddress, order.ShipAddress);
+            Assert.Equal(trackedOrder.ShipCity, order.ShipCity);
+            Assert.Equal(trackedOrder.ShipCountry, order.ShipCountry);
+            Assert.Equal(trackedOrder.ShipName, order.ShipName);
+            Assert.Equal(trackedOrder.ShippedDate, order.ShippedDate);
+            Assert.Equal(trackedOrder.ShipRegion, order.ShipRegion);
+            Assert.Equal(trackedOrder.ShipVia, order.Freight);
```

Comment:
Is this intended or should it be
```suggestion
            Assert.Equal(trackedOrder.ShipVia, order.ShipVia);
```
?